backend.py

```diff
@@ -65,10 +80,19 @@ def walk(self, distance, state, direction=None):
                     if state.robots[robot_in_the_way].coordinates != next_coordinates:
                         # Robot walks to new coordinates.
                         self.coordinates = next_coordinates
+                        # Check hole on new coordinates
+                        for tile in state.get_tiles(self.coordinates):
```

Comment:
Nebylo by lepsi zabalit ten kod (3x se opakujici) do funkce "check_hole" obdobne, jako to mame ted s s "check_wall"?